plugins/modules/katello_repository_set.py

```diff
@@ -159,10 +166,14 @@ def main():
             name=dict(),
             product=dict(),
             label=dict(),
-            repositories=dict(required=True, type='list', elements='dict'),
+            repositories=dict(type='list', elements='dict'),
+            all_repositories=dict(type='bool', default=False),
             state=dict(default='enabled', choices=['disabled', 'enabled']),
         ),
         required_one_of=[['label', 'name']],
+        required_if=[
+            ['all_repositories', False, ['repositories']],
```

Comment:
I'm a bit rusty, but do we also need one in the opposite direction?

```py
['repositories', [], ['all_repositories']]
```

Reply:
That would still allow the user to pass `all_repositories: False`, but yeah, this might be a good idea. I'll see that I add it.

Reply:
Ah, no, `all_repositories` defaults to `False`, so this never triggers. Let's see...

Reply:
And if I drop the default, it also doesn't work :(

Reply:
so the only way to trick the logic currently is to explicitly pass `repositories: []` which I think is safe enough?

Reply:
Hmm... but if you pass `repositories: []` and `enabled: true` will this module ensure no repositories are enabled?

Reply:
And if so, that makes me think we don't need an `all_repositories` setting - it'd be the same as `repositories: []` and `disabled: true`

Reply:
You mean state=disabled? Yeah, that'd probably work, but really not intuitive ;)

Reply:
Ah, no, repositories selects the "actionable set", so if you pass an empty list, the module does nothing.

Reply:
Ok great - so setting `repositories=dict(type='list', elements='dict', default=[])` should be fine and help solve the validation here, right?

Reply:
So I think I managed to confuse both you and me pretty well :crying_cat_face: 

When we set a default, `required_if` doesn't really work anymore, as it says "when field A is set to value B require field C to be set", but as C has a default, it's always set.

I have now dropped the defaults and added a `required_one_of` to the mix. What do you think?

Reply:
Yes, well that makes a lot more sense just looking at it ;) :+1: Fair warning: I didn't test it to be sure